code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/LoadBalancerDeploymentUnitInstance.java

```diff
@@ -0,0 +1,113 @@
+import io.cattle.iaas.lb.service.LoadBalancerService;
+import io.cattle.platform.core.constants.CommonStatesConstants;
+import io.cattle.platform.core.constants.InstanceConstants;
+import io.cattle.platform.core.constants.LoadBalancerConstants;
+import io.cattle.platform.core.model.Instance;
+import io.cattle.platform.core.model.LoadBalancer;
+import io.cattle.platform.core.model.LoadBalancerHostMap;
+import io.cattle.platform.core.model.Service;
+import io.cattle.platform.lb.instance.service.LoadBalancerInstanceManager;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.process.ObjectProcessManager;
+import io.cattle.platform.object.resource.ResourceMonitor;
+import io.cattle.platform.object.resource.ResourcePredicate;
+import io.cattle.platform.servicediscovery.api.constants.ServiceDiscoveryConstants;
+import io.cattle.platform.servicediscovery.deployment.DeploymentUnitInstance;
+import io.cattle.platform.servicediscovery.service.ServiceDiscoveryService;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class LoadBalancerDeploymentUnitInstance extends DeploymentUnitInstance {
+    LoadBalancerService lbService;
+    LoadBalancerInstanceManager lbInstanceMgr;
+
+    LoadBalancerHostMap hostMap;
+
+    public LoadBalancerDeploymentUnitInstance() {
+        super(null, null, null, null, null, null);
+    }
+
+    public LoadBalancerDeploymentUnitInstance(String uuid, Service service,
+            LoadBalancerHostMap hostMap, LoadBalancerService lbService,
+            LoadBalancerInstanceManager lbInstanceMgr, ObjectManager objectManager,
+            ResourceMonitor resourceMonitor, ObjectProcessManager objectProcessManager, ServiceDiscoveryService sdService) {
+        super(uuid, service, objectManager, resourceMonitor, objectProcessManager, sdService);
+        this.lbService = lbService;
+        this.lbInstanceMgr = lbInstanceMgr;
+        this.hostMap = hostMap;
+        if (hostMap != null) {
+            Instance instance = lbInstanceMgr.getLoadBalancerInstance(this.hostMap);
+            if (instance != null) {
+                this.instance = instance;
+            }
+        }
+    }
+
+    @Override
+    public boolean isError() {
+        if (this.hostMap.getState().equals(CommonStatesConstants.ACTIVE)) {
+            if (this.instance == null || this.instance.getRemoved() != null) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void remove() {
+        objectProcessManager.scheduleProcessInstance(LoadBalancerConstants.PROCESS_LB_HOST_MAP_REMOVE,
+                objectManager.reload(hostMap),
+                null);
+    }
+
+    @Override
+    public DeploymentUnitInstance start(List<Integer> volumesFromInstancesIds) {
+        if (this.hostMap == null) {
+            LoadBalancer lb = objectManager.findOne(LoadBalancer.class, LOAD_BALANCER.SERVICE_ID, service.getId(),
```

Comment:
You have to be careful with findOne, if it finds two, it blows up with an exception.  Where as findAny just takes the first.  Sometime you have to decide if you want it to blow up or does something slightly wrong...  Not an easy decision a lot of the time.